assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+
+.Procedure
+
+. Add the bridge dependency to the project `pom.xml` file:
++
+[source,xml,options="nowrap"]
+----
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+      <version>1.x</version>
+    </dependency>
+----
+
+. Configure the logging level on the package `org.optaplanner` in your `log4j.xml` file as shown in the following example, where `_LEVEL_` is a logging level listed in XREF:
++
+[source,xml,options="nowrap"]
+----
+<log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
+
+  <category name="org.optaplanner">
+    <priority value="_LEVEL_" />
+  </category>
+
+  ...
+
+</log4j:configuration>
+----
+
+. Optional: If you have a multitenant application where multiple `Solver` instances might be running at the same time, separate the logging of each instance into separate files:
++
+COMMENT: this section is about LOG4J so can we have a LOG4J example here?
++
+.. Surround the `solve()` call with  http://logback.qos.ch/manual/mdc.html[Mapped Diagnostic Context] (MDC):
++
+[source,java,options="nowrap"]
+----
+        MDC.put("tenant.name",tenantName);
+        MySolution bestSolution = solver.solve(problem);
+        MDC.remove("tenant.name");
+----
+
+.. Configure your logger to use different files for each ``${tenant.name}``.
+For example in Logback, use a `SiftingAppender` in ``logback.xml``:
++
+[source,xml,options="nowrap"]
+----
+  <appender name="fileAppender" class="ch.qos.logback.classic.sift.SiftingAppender">
+    <discriminator>
+      <key>tenant.name</key>
+      <defaultValue>unknown</defaultValue>
+    </discriminator>
+    <sift>
+      <appender name="fileAppender.${tenant.name}" class="...FileAppender">
+        <file>local/log/optaplanner-${tenant.name}.log</file>
+        ...
+      </appender>
+    </sift>
+  </appender>
+----
+
+
+
+[[randomNumberGenerator]]
+=== Configuring the random number generator
+
+Many heuristics and metaheuristics depend on a pseudorandom number generator for move selection, to resolve score ties, probability based move acceptance, and so on. During solving, the same `Random` instance is reused to improve reproducibility, performance, and uniform distribution of random values.
+
+A random seed is a number used to initialize a pseudorandom number generator.
+
+.Procedure
+. Optional: To change the random seed of that `Random` instance, specify a ``randomSeed``:
++
+[source,xml,options="nowrap"]
+----
+<solver xmlns="https://www.optaplanner.org/xsd/solver" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://www.optaplanner.org/xsd/solver https://www.optaplanner.org/xsd/solver/solver.xsd">
+  <randomSeed>0</randomSeed>
+  ...
+</solver>
+----
+
+. Optional: To change the pseudorandom number generator implementation, specify a value for the ``randomType`` property,where `_RANDOM_NUMBER_GENERATOR_` is a pseudorandom number generator listed below:
++
+COMMENT: Is this the configuration file?
```

Comment:
Yes it is solver config file the one which u mention when create solver(for example nqueensSolverConfig.xml) 

```
        SolverFactory<NQueens> solverFactory = SolverFactory.createFromXmlResource(
                "nqueensSolverConfig.xml");
        Solver<NQueens> solver = solverFactory.buildSolver();
```

Reply:
This comment was outdated. I had already updated the text with clarifications from Christopher.